### PR TITLE
Update to Honeycrisp 0.10.5 to 0.15.2

### PR DESCRIPTION
**Pull request by arinchoi03**
## Link to pivotal/JIRA issue
- [TBE-118](https://codeforamerica.atlassian.net/browse/TBE-118)

## Is PM acceptance required? (delete one)
- Yes - don't merge until JIRA issue is accepted!

**Reminder**: merge main into this branch and get green tests before merging to main 

## What was done?
- Honeycrisp was updated to the most recent version ✨ 
- New updates
- I tried to keep the tax-specific versions of the methods in place while bringing over the updates from honeycrisp
- the hub "reassign user" reveal had a visual change (but the reveals on the state flows looked the same) so I tweaked styling there. 
- Unfortunately, it's difficult to test visual diffs b/c the main branch does not screenshot a lot of the state file pages. Should we add it?

## How to test?
- Describe the testing approach taken to verify the changes, including:
  - Unit/integration/manual tests
  - Test data used
- Specify any relevant testing environments used (e.g., development, staging, demo, Heroku).
- Risk Assessment
  - Risks or side effects associated with the changes and how they were mitigated.
  - Highlight areas that may need extra attention during code review or testing.
  - Paste SQL queries or output where relevant
## Screenshots (for visual changes)
- Before
- After

[TBE-118]: https://codeforamerica.atlassian.net/browse/TBE-118?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by arinchoi03** [comment]
There are some visual diffs (I think I need to add more screenshots to test b/c I wasn't including many of the pages in the previous percy run)  

with `.reveal__button`
Screenshot 2024-10-18 at 3 45 20 PM

with `.reveal__link`
Screenshot 2024-10-18 at 3 54 19 PM

Also I'm finding that we have a local `app/javascript/lib/honeycrisp.js` file in our repo with code that looks incredibly similar to `app/assets/javascripts/honeycrisp.js` in the honeycrisp repo...I'm guessing we're just pasting the code into there? I'm a bit confused by this set up but trying to figure it out.

File: app/javascript/lib/honeycrisp.js [hunk 1]
```diff
@@ -98,28 +106,32 @@ var followUpQuestion = (function() {
 
                 // set initial state of follow-ups based on the page
                 $(this).find('input').each(function(index, input) {
-                    if($(this).is(':checked') && $(this).attr('data-follow-up') != null) {
-                        $($(this).attr('data-follow-up')).show();
+                    if($(this).attr('data-follow-up') != null) {
+                        $($(this).attr('data-follow-up')).toggle($(this).is(':checked'));
```
Comment (arinchoi03) [inline app/javascript/lib/honeycrisp.js:110]: added this back https://github.com/codeforamerica/vita-min/commit/fa6ea89ab04d87fd7091ddc2f028d7808adcb266#diff-395ae426475fee69171cc850fa90abec0b8c28872fdb039818382a2a40aaa327

File: app/javascript/lib/honeycrisp.js [hunk 2]
```diff
@@ -98,28 +106,32 @@ var followUpQuestion = (function() {
...
+                    $container.find('.question-with-follow-up__follow-up input').attr('disabled', false);
+                    $($(this).attr('data-follow-up')).show();
+                }
             });
         }
     }
     return {
-        init: fUQ.init
+        init: fUQ.init,
+        update: fUQ.update
```
Comment (arinchoi03) [inline app/javascript/lib/honeycrisp.js:134]: this `update` change is a result of https://github.com/codeforamerica/honeycrisp-gem/commit/c05fb0b74c1ecf67a62d9735c5dc77240376beb2

File: app/javascript/lib/honeycrisp.js [hunk 3]
```diff
@@ -129,9 +141,19 @@ var revealer = (function() {
...
+                })
+                revealButton.click(function(e) {
                     e.preventDefault();
                     $(self).toggleClass('is-hiding-content');
+
+                    if (this.getAttribute('aria-expanded') === 'false') {
+                        this.setAttribute('aria-expanded', 'true');
+                    } else {
+                        this.setAttribute('aria-expanded', 'false')
+                    }
```
Comment (arinchoi03) [inline app/javascript/lib/honeycrisp.js:156]: reveal accessibility updates: https://github.com/codeforamerica/honeycrisp-gem/pull/317

File: app/javascript/lib/honeycrisp.js [hunk 4]
```diff
@@ -216,7 +225,15 @@ var noneOfTheAbove = (function() {
             $noneCheckbox.click(function(e) {
                 $otherCheckboxes.prop('checked', false);
                 $otherCheckboxes.parent().removeClass('is-selected');
+
+                // If we just unchecked an <input> with a follow-up, let's reset the follow-up questions
+                // so it hides properly.
+                var $enclosingFollowUp = $noneCheckbox.closest('.question-with-follow-up');
+                if ($enclosingFollowUp) {
+                    followUpQuestion.update($enclosingFollowUp);
+                }
```
Comment (arinchoi03) [inline app/javascript/lib/honeycrisp.js:234]: also part of https://github.com/codeforamerica/honeycrisp-gem/commit/c05fb0b74c1ecf67a62d9735c5dc77240376beb2

File: app/javascript/lib/honeycrisp.js [hunk 5]
```diff
@@ -55,8 +55,16 @@ var radioSelector = (function() {
...
+                    var allRadioButtons;
+                    if ($(this).closest('.cfa-radio-button').length > 0) {
+                        // FormBuilder V2
+                        allRadioButtons = $(this).closest('.cfa-radio-button').siblings().children('.is-selected');
+                    } else {
+                        // FormBuilder V1 & GCF
+                        allRadioButtons = $(this).closest('.radio-button').siblings();
+                    }
+                    allRadioButtons.removeClass('is-selected');
+                    $(this).closest('.radio-button').addClass('is-selected');
```
Comment (arinchoi03) [inline app/javascript/lib/honeycrisp.js:67]: implementation looks same to me, but just accommodates V2, if ever used

File: app/javascript/lib/honeycrisp.js [hunk 6]
```diff
@@ -104,22 +112,26 @@
...
+        update: function ($container){
+            // reset follow ups
+            $container.find('.question-with-follow-up__follow-up input').attr('disabled', true);
+            $container.find('.question-with-follow-up__follow-up').hide();
+
+            $container.find('.question-with-follow-up__question input').each(function(index, input) {
+                // if any of the inputs with a data-follow-up is checked then show the follow-up
+                if($(input).is(':checked') && $(input).attr('data-follow-up') != null) {
+                    $container.find('.question-with-follow-up__follow-up input').attr('disabled', false);
+                    $($(this).attr('data-follow-up')).show();
```
[the remark this answers is not in the document]
Comment (arinchoi03) [reply app/javascript/lib/honeycrisp.js:127]: how do we feel about autofix? Also this is a straightup copy-paste from honeycrisp, maybe we should update that too.

**Comment by arinchoi03** [comment]
running percy locally (not against main b/c it doesn't have the screenshots I want)

File: app/views/state_file/questions/nj_homeowner_property_tax/edit.html.erb [hunk 7]
```diff
@@ -12,7 +12,7 @@
     </div>
 
     <div class="reveal">
-      <p><a href="#" class="reveal__link"><%= t('.helper_heading') %></a></p>
+      <p><button class="reveal__button"><%= t('.helper_heading') %></button></p>
```
Comment (mmazanec22) [inline app/views/state_file/questions/nj_homeowner_property_tax/edit.html.erb:15]: Do we need to add an aria-expanded value to all of these reveal buttons? Or is that handled somewhere else?

Comment (arinchoi03) [reply app/views/state_file/questions/nj_homeowner_property_tax/edit.html.erb:15]: @mmazanec22 `app/assets/javascripts/honeycrisp.js` has code for this in `revealer` part of the code where it sets `aria-expanded` to `false`, but if you're not seeing that get set, please let me know! I can double check this on Monday when I'm back

Comment (mmazanec22) [reply app/views/state_file/questions/nj_homeowner_property_tax/edit.html.erb:15]: It does appear to be working! Nice, good to know we don't need to manually add anything

File: app/javascript/lib/honeycrisp.js (hunk 6 above)
[the remark this answers is not in the document]
Comment (arinchoi03) [reply app/javascript/lib/honeycrisp.js:127]: discussed in standup, will make the necessary changes to address this & suggest update to honeycrisp main repo

**Comment by embarnard** [comment]
@arinchoi03 re:percy, did you try running `bin/percy` locally?

**Comment by arinchoi03** [comment]
@embarnard yes it does run! but just doesn't on the FYST pages I marked locally (or it does but doesn't show any changes since the main branch doesn't take screenshots of those pages)

File: app/assets/stylesheets/_state-file.scss [hunk 8]
```diff
@@ -516,7 +516,7 @@
     }
   }
 
-  .reveal__link {
+  .reveal__button {
```
Comment (embarnard) [inline app/assets/stylesheets/_state-file.scss:519]: I still see some instances of `reveal__link` in the code base, should we change these? @arinchoi03 
Screenshot 2024-10-28 at 4 14 49 PM

File: app/javascript/lib/honeycrisp.js (hunk 5 above)
Comment (embarnard) [reply app/javascript/lib/honeycrisp.js:67]: I think its fine to leave in 🤷🏻‍♀️

File: app/assets/stylesheets/_state-file.scss (hunk 8 above)
Comment (arinchoi03) [reply app/assets/stylesheets/_state-file.scss:519]: ooh these must be new ones that got merged in from `main`. I'll fix!

**Review by embarnard — APPROVED** [review]
Great work! I wouldn't mind adding in FYST screenshots into percy but don't think its absolutely necessary right now if it constitutes a large chunk of work. I think it would look something like adding `screenshot: true` to each state flow complete intake feature test scenario line declaration like we see in `new_joint_filers_spec.rb` and wrapping page assertions with `screenshot after` blocks which might be more annoying. I don't think its a blocker but maybe a nice to have for the future so I'll let you determine whether or not to do it now or save it for later. In any case thanks for doing this work, approved! ✅

**Comment by arinchoi03** [comment]
**NOTE** Before merging, make sure to merge in main & scan for any new `reveal__link` elements in the codebase!!!

**Pull request merged**